testdata/src/default_config/fix_cuddle_whole_blocks/fix_cuddle_whole_blocks.go

```diff
@@ -0,0 +1,133 @@
+package testpkg
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func ForCuddleAssignmentWholeBlock() {
+	x := 1
+	y := 2
+
+	var numbers []int
+	for i := 0; i < 10; i++ {
+		if x == y {
+			numbers = append(numbers, i)
+		}
+	}
+
+	var numbers2 []int
+	for i := range 10 {
+		if x == y {
+			numbers2 = append(numbers2, i)
+		}
+	}
+
+	var numbers3 []int
+	for { // want "for statement without condition should never be cuddled"
+		if x == y {
+			numbers3 = append(numbers3, i)
+		}
+	}
+
+	environment = make(map[string]string)
+	for _, env := range req.GetConfig().GetEnvs() {
+		switch env.GetKey() {
+		case "user-data":
+			cloudInitUserData = env.GetValue()
+		default:
+			environment[env.GetKey()] = env.GetValue()
+		}
+	}
+}
+
+func IfCuddleAssignmentWholeBlock() {
+	x := 1
+	y := 2
+
+	counter := 0
+	if somethingTrue {
+		checker := getAChecker()
+		if !checker {
+			return
+		}
+
+		counter++ // Cuddled variable used in block, but not as first statement
+	}
+
+	var number2 []int
+	if x == y {
+		fmt.Println("a")
+	} else {
+		if x > y {
+			number2 = append(number2, i)
+		}
+	}
+
+	var number3 []int
+	if x == y {
+		fmt.Println("a")
+	} else if x > y {
+		if x == y {
+			number3 = append(number3, i)
+		}
+	}
+
+	var number4 []int
+	if x == y {
+		if x == y {
+			number4 = append(number4, i)
+		}
+	} else if x > y {
+		if x == y {
+			number4 = append(number4, i)
+		}
+	} else {
+		if x > y {
+			number4 = append(number4, i)
+		}
+	}
+}
+
+func SwitchCuddleAssignmentWholeBlock() {
+	var id string
+	var b bool // want "declarations should never be cuddled"
+	switch b { // want "only one cuddle assignment allowed before switch statement"
```

Comment:
What happens if we put them in the other order, that should only yield the declaration error, right? Can we add a test for  it and assert that it gets fixed properly?

```go
var b bool
var id string
switch b {
```

Should be come

```go
var b bool

var id string
switch b {
```